pkgs/development/python-modules/flask-basicauth/default.nix

```diff
@@ -0,0 +1,35 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, fetchpatch
+, flask
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "Flask-BasicAuth";
```

Comment:
The canonical repo name on GitHub is all lowercase -- perhaps it would also be appropriate to convert it here? I see a few Flask packages use uppercase, but we conventionally use all lowercase `pname`s. If you feel differently, let me know, happy to discuss.

Reply:
thanks, i updated it to be lowercase. i think i was experimenting with `fetchPypi` at one point, and i had made the name capitalized then to match what's on pypi. i agree that the lowercase feels more correct.

Reply:
Names should adhere to what is described in PEP503, see https://peps.python.org/pep-0503/#normalized-names.

> This PEP references the concept of a “normalized” project name. As per PEP 426 the only valid characters in a name are the ASCII alphabet, ASCII numbers, ., -, and _. The name should be lowercased with all runs of the characters ., -, or _ replaced with a single - character. This can be implemented in Python with the re module: